ear-production-suite-plugins/plugins/scene/src/scene_plugin_processor.cpp

```diff
@@ -125,15 +125,15 @@ bool SceneAudioProcessor::isBusesLayoutSupported(
 
 void SceneAudioProcessor::processBlock(AudioBuffer<float>& buffer,
                                        MidiBuffer& midiMessages) {
+
+  metadataThread_.post([this](){
+    backend_->triggerMetadataSend();
```

Comment:
Since this is put in a queue, we can not guarantee it will execute before the monitoring plugins receive the block. Possibly not a problem since the timing is probably imperceptible?

Reply:
We can guarantee the renderers will get metadata in the correct order as only one thread does metadata sending and the triggers are processed FIFO. We can't guarentee exactly when they'll get them as it depends when the sending thread gets around to it, but this was already true as they were sent and received using nng async io (on an nng worker thread). I don't think there's a way to make sure the renderers see the metadata on the same audio block as we don't even know which order the plugins will get their audio callback.

Reply:
I guess we could report some latency in the input and scene plugins, so plugin delay compensation would mean they were processed earlier, then timestamp the metadata so it's not read until the correct block, but I think we decided that wasn't necessary way back when..